packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/utils.ts

```diff
@@ -0,0 +1,793 @@
+					}
+				} else {
+					returnData = responseData;
+				}
+
+				if (fieldsToInclude === 'selected') {
+					for (const item of responseData) {
+						const newItem: IDataObject = {};
+
+						for (const field of fields) {
+							set(newItem, field, get(item, field));
+						}
+
+						returnData.push(newItem);
+					}
+				}
+
+				if (fieldsToInclude === 'except') {
+					for (const item of responseData) {
+						for (const field of fields) {
+							unset(item, field);
+						}
+
+						returnData.push(item);
+					}
+				}
+
+				return JSON.stringify(returnData, null, 2);
+			};
+		}
+	}
+
+	return <T>(response: T) => {
+		if (typeof response === 'string') {
+			return response;
+		}
+		if (typeof response === 'object') {
+			return JSON.stringify(response, null, 2);
+		}
+
+		return String(response);
+	};
+};
+
+const extractPlaceholders = (text: string): string[] => {
+	const placeholder = /(\{[a-zA-Z0-9_]+\})/g;
+	const returnData: string[] = [];
+
+	const matches = text.matchAll(placeholder);
+
+	for (const match of matches) {
+		returnData.push(match[0].replace(/{|}/g, ''));
+	}
+
+	return returnData;
+};
+
+export const extractParametersFromText = (
+	placeholders: PlaceholderDefinition[],
+	text: string,
+	sendIn: SendIn,
+	key?: string,
+): ToolParameter[] => {
+	if (typeof text !== 'string') return [];
+
+	const parameters = extractPlaceholders(text);
+
+	if (parameters.length) {
+		// eslint-disable-next-line @typescript-eslint/no-use-before-define
+		const inputParameters = prepareParameters(
+			parameters.map((name) => ({
+				name,
+				valueProvider: 'modelRequired',
+			})),
+			placeholders,
+			'keypair',
+			sendIn,
+			'',
+		);
+
+		return key
+			? inputParameters.parameters.map((p) => ({ ...p, key }))
+			: inputParameters.parameters;
+	}
+
+	return [];
+};
+
+function prepareParameters(
+	rawParameters: RawParametersValues,
+	placeholders: PlaceholderDefinition[],
+	parametersInputType: 'model' | 'keypair' | 'json',
+	sendIn: SendIn,
+	modelInputDescription: string,
+	jsonWithPlaceholders?: string,
+): { parameters: ToolParameter[]; values: IDataObject } {
+	const parameters: ToolParameter[] = [];
+	const values: IDataObject = {};
+
+	if (parametersInputType === 'model') {
+		return {
+			parameters: [
+				{
+					name: sendIn,
+					required: true,
+					type: 'json',
+					description: modelInputDescription,
+					sendIn,
+				},
+			],
+			values: {},
+		};
+	}
+
+	if (parametersInputType === 'keypair') {
+		for (const entry of rawParameters) {
+			if (entry.valueProvider.includes('model')) {
+				const placeholder = placeholders.find((p) => p.name === entry.name);
+
+				const parameter: ToolParameter = {
+					name: entry.name,
+					required: entry.valueProvider === 'modelRequired',
+					sendIn,
+				};
+
+				if (placeholder) {
+					parameter.type = placeholder.type;
+					parameter.description = placeholder.description;
+				}
+
+				parameters.push(parameter);
+			} else {
```

Comment:
```suggestion
			}  else if (entry.value) {
```
If we check for `entry.value` here we don't need to do the type coercion on line 483